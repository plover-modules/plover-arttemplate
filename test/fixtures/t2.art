{{// 转义输出 }}
{{name}}

{{// 非转义输入 }}
{{=version}}

{{// 判断语句}}
{{if tag == 'a'}}
  A
{{else if tag == 'b'}}
  B
{{else}}
  C
{{/if}}

{{// 循环语句 }}
{{each keywords as keyword}}
  {{keyword}}
{{/each}}

{{// 空值 }}
{{=null}}
{{=undefined}}
{{null}}
{{undefined}}


{{// 纯语句单行 }}
{{! var a = function() { return '123' } }}

{{a}}
{{=a}}

{{o}}
{{=o}}

{{123}}
{{=123}}

{{// 纯语句多行 }}
{{!
var k = 123
}}

{{ k + 1 }}

{{// map迭代 }}
{{each map as v k}}
  {{k}}{{v}}
{{/each}}

{{// 空值迭代不报错 }}
{{each null as index }}
{{/each}}


{{// 我是单行注释}}

{{//
我是多行注释
我是多行注释
我是多行注释
}}


{{// 输出特殊字符 }}
<div class="item-\{{ id \}}">
  a=\{{ a \}}, b=\{{ b \}}
</div>


{{// helper }}
{{format(123, '元')}}

{{// range }}
{{each $range(5) as index }}
  {{index}}
{{/each}}


{{! var list = $range(5, 1, 2) }}
{{each list as item}}
{{item}}
{{/each}}


{{!
  // 注释在语句中
  var a = 1;
  if (!showIndexInPage < 2) {
    // 首屏直接显示220图片
    a = 2;
  }
}}
{{a}}
