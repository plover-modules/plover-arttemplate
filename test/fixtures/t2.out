
Plover
1.0.1

B

framework
koa
es6

123
123

<h1>hello world</h1>
<h1>hello world</h1>

123
123

124

namePlover
version0.1.0

<div class="item-{{ id }}">
  a={{ a }}, b={{ b }}
</div>

123å…ƒ

0
1
2
3
4

1
3
5
7
9

2
